centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala

```diff
@@ -41,18 +41,20 @@ object TestFormulas extends StrictLogging {
   def runSuccessfulWorkflowAndVerifyMetadata(workflowDefinition: Workflow)(implicit cromwellTracker: Option[CromwellTracker]): Test[SubmitResponse] = for {
     _ <- checkDescription(workflowDefinition, validityExpectation = Option(true))
     submittedWorkflow <- runSuccessfulWorkflow(workflowDefinition)
-    metadata <- validateMetadata(submittedWorkflow, workflowDefinition)
-    _ <- validateJobManagerStyleMetadata(submittedWorkflow, metadata.value)
-    _ = cromwellTracker.track(metadata)
-    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, metadata)
+    notArchivedMetadata <- validateMetadata(submittedWorkflow, workflowDefinition, validateArchived = Option(false))
+    _ <- validateJobManagerStyleMetadata(submittedWorkflow, originalMetadata = notArchivedMetadata.value, validateArchived = Option(false))
+    _ = cromwellTracker.track(notArchivedMetadata)
+    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, notArchivedMetadata)
     _ <- waitForArchive(submittedWorkflow, workflowDefinition)
     // Re-validate the metadata now that carboniting has completed
-    _ <- validateMetadata(submittedWorkflow, workflowDefinition)
-    flatMetadata = metadata.asFlat
-    workflowRoot = flatMetadata.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
+    _ <- validateMetadata(submittedWorkflow, workflowDefinition, validateArchived = Option(true))
+    _ <- validateJobManagerStyleMetadata(submittedWorkflow, originalMetadata = notArchivedMetadata.value, validateArchived = Option(true))
+    _ <- validateSubworkflowsMetadataBeforeAndAfterArchival(submittedWorkflow, workflowDefinition)
+    notArchivedFlatMetadata = notArchivedMetadata.asFlat
+    workflowRoot = notArchivedFlatMetadata.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
     _ <- validateOutputs(submittedWorkflow, workflowDefinition, workflowRoot)
     _ <- validateLabels(submittedWorkflow, workflowDefinition, workflowRoot)
-    _ <- validateLogs(metadata, submittedWorkflow, workflowDefinition)
+    _ <- validateLogs(notArchivedMetadata, submittedWorkflow, workflowDefinition)
```

Comment:
The ticket AC includes checking that these "other metadata-based endpoints" don't change pre-vs-post carboniting in its AC. Since that probably implies work to add the `metadataSource` parameter option to these other endpoints, you could argue to bump that work to a follow up ticket/PR, but sooner or later we'll need to do it.